package.json

```diff
@@ -30,7 +30,8 @@
     "format": "purs-tidy format-in-place core src bin docs-search",
     "format:check": "purs-tidy check core src bin docs-search",
     "bundle": "spago bundle -p spago-bin",
-    "prepublishOnly": "spago build && ./bin/index.dev.js bundle -p spago-bin && ./bin/index.dev.js bundle -p docs-search-client-halogen"
+    "prepublishOnly": "spago build && ./bin/index.dev.js bundle -p spago-bin && ./bin/index.dev.js bundle -p docs-search-client-halogen",
+    "halogen:dev": "echo \"#!/usr/bin/env node\n\nimport { main } from './output/Docs.Search.App/index.js'; main();\" | esbuild --platform=browser --format=iife --sourcemap --bundle --outfile=generated-docs/html/docs-search-app.js --loader:.node=file --alias:punycode=punycode/ --watch=forever"
```

Comment:
A few notes:
- `spago bundle` only allows `extraArgs` in the spago.yaml configs, not from the command line. So I was forced to use esbuild directly. 
- Esbuild doesn't know its supposed to invoke the `main` function so I copied the same workarounds from `Spago.Command.Docs.` i.e. it needs the --iife flag, and the main function to be invoked via extra stdin input.
- --watch=forever is so that it keeps watching even after stdin stops receiving input. This does the right thing ™️ , it rebuilds when output/ dependencies change. If you run in regular watch mode, you get this error:

```console
[watch] build finished, watching for changes...
[watch] stopped because stdin was closed (use "--watch=forever" to keep watching even after stdin is closed)
```

Reply:
> only allows `extraArgs` in the spago.yaml configs, not from the command line

Oh, that looks like an oversight 😄 I think we have a good argument to add this now

Reply:
> it needs the --iife flag

Why does it need that?

Reply:
> Why does it need that?

To bundle for the browser, [see esbuild docs](https://esbuild.github.io/api/#platform)

Reply:
:thinking: Wasn't there some other reason though about why we couldn't use ESM specifically? I thought there was some library that had an issue, but I can't recall which or why.